src/workerd/jsg/resource.h

```diff
@@ -191,6 +192,125 @@ template <typename TypeWrapper,
+  static constexpr bool value = IsFastApiCompatibleReturnType<T>::value ||
+      std::is_same_v<T, v8::Local<v8::Value>> || std::is_same_v<T, v8::Local<v8::Object>>;
+};
+
+// Helper to determine if a method can be used with V8 fast API
+template <typename Ret, typename... Args>
+struct IsFastApiCompatible {
+  static constexpr bool value = IsFastApiCompatibleReturnType<Ret>::value &&
+      (IsFastApiCompatibleParamType<Args>::value && ...);
+};
+
+// Helper to determine if a method pointer type is compatible with Fast API
+template <typename Method>
+struct IsFastMethodCompatible {
+  static constexpr bool value = false;
+};
+
+// Specialization for non-static member functions
+template <typename Class, typename Ret, typename... Args>
+struct IsFastMethodCompatible<Ret (Class::*)(Args...)> {
+  static constexpr bool value = IsFastApiCompatible<Ret, Args...>::value;
+};
+
+// Specialization for const non-static member functions
+template <typename Class, typename Ret, typename... Args>
+struct IsFastMethodCompatible<Ret (Class::*)(Args...) const> {
+  static constexpr bool value = IsFastApiCompatible<Ret, Args...>::value;
+};
+
+// Implements V8 Fast API callback for a method. This class handles both the fast and slow paths.
+template <typename TypeWrapper,
+    const char* methodName,
+    bool isContext,
+    typename T,
+    typename Method,
+    Method method,
+    typename Indexes>
+struct FastMethodCallback;
+
+// Specialization for instance methods compatible with Fast API
+template <typename TypeWrapper,
+    const char* methodName,
+    bool isContext,
+    typename T,
+    typename U,
+    typename Ret,
+    typename... Args,
+    Ret (U::*method)(Args...),
+    size_t... indexes>
+struct FastMethodCallback<TypeWrapper,
+    methodName,
+    isContext,
+    T,
+    Ret (U::*)(Args...),
+    method,
+    kj::_::Indexes<indexes...>> {
+
+  template <bool trackCallCount = false>
+  static void slowCallback(const v8::FunctionCallbackInfo<v8::Value>& args) {
+    if constexpr (trackCallCount) {
+      jsg::fast_api::callCounter.slow++;
+    }
+    MethodCallback<TypeWrapper, methodName, isContext, T, Ret (U::*)(Args...), method,
+        kj::_::Indexes<indexes...>>::callback(args);
+  }
+
+  template <bool trackCallCount = false>
+  static Ret fastCallback(
+      v8::Local<v8::Object> receiver, Args... fastArgs, v8::FastApiCallbackOptions& options) {
+
+    if constexpr (trackCallCount) {
+      jsg::fast_api::callCounter.fast++;
+    }
+    auto isolate = options.isolate;
+    v8::HandleScope handleScope(isolate);
+    auto context = isolate->GetCurrentContext();
+    auto& self = extractInternalPointer<T, isContext>(context, receiver);
+
+    try {
+      if constexpr (std::is_same_v<Ret, void>) {
+        (self.*method)(fastArgs...);
+        return;
+      } else {
+        return (self.*method)(fastArgs...);
+      }
+    } catch (kj::Exception& exception) {
+      throwInternalError(isolate, kj::mv(exception));
```

Comment:
If you notice in the implementation of `liftKj(...)` we actually use a nested try catch pattern. Are you certain that the same pattern is not needed here? cc'ing @kentonv as the author of liftKj